modules/packages/HDF5.chpl

```diff
@@ -3784,11 +3784,134 @@ module HDF5 {
     C_HDF5.H5Fclose(file_id);
   }
 
+  /* Write the Block distributed Array `A` as an HDF5 dataset named `dsetName`
+     in the file `filename` using parallel collective IO. This requires the
+     hdf5-parallel library, which requires the MPI library.
+
+     The file written by this function can be read in parallel with the
+     function `hdf5ReadDistributedArray`. The write and read operations
+     can use arrays distributed over different numbers of locales.
+   */
+  proc hdf5WriteDistributedArray(A: [], filename: string, dsetName: string) {
```

Comment:
In today's demo, I found myself thinking "What's the point of using an hdf5 prefix on this function given that it's defined in the HDF5 module?"  It seems like a very flat, C-based naming scheme for a language that doesn't need it.